airbyte-integrations/connectors/destination-databricks/src/main/kotlin/io/airbyte/integrations/destination/databricks/operation/DatabricksStorageOperation.kt

```diff
@@ -116,11 +106,52 @@ class DatabricksStorageOperation(
         workspaceClient.files().createDirectory(stagingDirectory(streamId, database))
     }
 
-    override fun prepareStage(streamId: StreamId, destinationSyncMode: DestinationSyncMode) {
-        prepareStagingTable(streamId, destinationSyncMode)
+    override fun prepareStage(streamId: StreamId, suffix: String, replace: Boolean) {
+        prepareStagingTable(streamId, suffix, replace)
         prepareStagingVolume(streamId)
     }
 
+    override fun overwriteStage(streamId: StreamId, suffix: String) {
+        // databricks recommends CREATE OR REPLACE ... AS SELECT
+        // instead of dropping the table and then doing more operations
+        // https://docs.databricks.com/en/delta/drop-table.html#when-to-replace-a-table
+        destinationHandler.execute(
+            // Databricks doesn't support transactions, so we have to do these separately
+            Sql.separately(
+                """
+                CREATE OR REPLACE TABLE `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}`
+                AS SELECT * FROM `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}$suffix`
+                """.trimIndent(),
+                "DROP TABLE `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}$suffix`",
+            )
+        )
+    }
+
+    override fun transferFromTempStage(streamId: StreamId, suffix: String) {
+        destinationHandler.execute(
+            // Databricks doesn't support transactions, so we have to do these separately
+            Sql.separately(
+                """
+                INSERT INTO `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}`
+                SELECT * FROM `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}$suffix`
+                """.trimIndent(),
+                "DROP TABLE `$database`.`${streamId.rawNamespace}`.`${streamId.rawName}$suffix`",
+            )
+        )
+    }
+
+    override fun getStageGeneration(streamId: StreamId, suffix: String): Long? {
```

Comment:
To satisfy my curiosity, do we pass any other suffixes through these functions besides `"_airbyte_tmp"` and `""`? Is this always really just `isTemp=true/false`?

Reply:
yep, that's correct for now. At some point I think we maybe should do generation-specific suffixes, but that's a whole different can of worms, and definitely not a consideration for the near future